docs/development/qubes_staging.rst

```diff
@@ -0,0 +1,313 @@
+
+Later we'll be using Ansible to provision the application VMs, so we should
+make sure we can SSH between those machines without needing to type
+a password. On ``sd-dev``:
+
+.. code:: sh
+
+   ssh-copy-id sdadmin@10.137.0.50
+   ssh-copy-id sdadmin@10.137.0.51
+
+Confirm that you're able to ssh as user ``sdadmin`` from ``sd-dev`` to
+``sd-mon-base`` and ``sd-app-base`` without being prompted for a password.
+
+SecureDrop Installation
+-----------------------
+
+We're going to configure ``sd-dev`` to build the SecureDrop ``.deb`` files,
+then we're going to build them, and provision ``sd-app`` and ``sd-mon``.
+Follow the instructions in the :doc:`developer documentation <setup_development>`
+to set up the development environment.
+
+Once finished, build the Debian packages for installation on the staging VMs.
+
+.. code::
+
+   make build-debs
+
+The ``.deb`` files will be available in ``build/``.
+
+Managing Qubes RPC for Admin API capability
+-------------------------------------------
+
+We're going to be running Qubes management commands on ``sd-dev``,
+which requires some additional software. Install it with
+
+.. code::  sh
+
+    sudo apt install qubes-core-admin-client
+
+You'll need to grant the ``sd-dev`` VM the ability to create other VMs,
+by editing the Qubes RPC policies in ``dom0``. Here is an example of a
+permissive policy, sufficient to grant ``sd-dev`` management capabilities
+over VMs it creates:
+
+.. todo::
+
+   Reduce these grants to the bare minimum necessary. We can likely
+   pare them down to a single grant, preferably with tags-based control.
+
+.. code:: sh
+
+   /etc/qubes-rpc/policy/include/admin-local-rwx:
+     sd-dev $tag:created-by-sd-dev allow,target=$adminvm
+
+   /etc/qubes-rpc/policy/include/admin-global-rwx:
+     sd-dev $adminvm allow,target=$adminvm
+     sd-dev $tag:created-by-sd-dev allow,target=$adminvm
+
+.. tip::
+
+   See the Qubes documentation for details on leveraging the `Admin API`_.
+
+.. _`Admin API`: https://www.qubes-os.org/doc/admin-api/
+
+Creating staging instance
+-------------------------
+
+After creating the StandaloneVMs as described above:
+
+* ``sd-dev``
+* ``sd-trusty-base``
+* ``sd-app-base``
+* ``sd-mon-base``
+
+And after building the SecureDrop .debs, we can finally provision the staging
+environment. In from the root of the SecureDrop project in ``sd-dev``, run:
+
+.. code:: sh
+
+   molecule test -s qubes-staging
+
+.. note::
+   The reboot actions run against the VMs during provisioning will only shutdown
+   the VMs, not start them again, since these are Standalone VMs. Therefore
+   the ``test`` action will fail by default, unless you judiciously run
+   ``qvm-start <vm>`` for each VM after they've shut down.
+
+You can use the smaller constituent Molecule actions, rather than the bundled
+``test`` action:
+
+.. code:: sh
+
+   molecule create -s qubes-staging
+   molecule prepare -s qubes-staging
```

Comment:
Is the prepare step necessary? The `molecule create` does call the prepare action after creating the VMs, and warns that there is no prepare playbook found (`molecule/qubes-staging/prepare.yml`). Running `molecule prepare` explicitly  returns a `Skipping, instances already prepared` message.